storagenode/pieces/store.go

```diff
@@ -250,6 +276,28 @@ func (store *Store) Delete(ctx context.Context, satellite storj.NodeID, pieceID
 	return Error.Wrap(err)
 }
 
+func (store *Store) getPieceSize(ctx context.Context, blobRef storage.BlobRef) (int64, error) {
+	blobAccess, err := store.blobs.Lookup(ctx, blobRef)
+	if err != nil {
+		return 0, err
+	}
+	pieceAccess, err := newStoredPieceAccess(store, blobAccess)
+	if err != nil {
+		return 0, err
+	}
+	return pieceAccess.ContentSize(ctx)
+}
+
+// UpdateLiveUsedSpaceTotals updates the live used space totals
+// with a pieceSize that was either created or deleted where the pieceSize is
+// only the content size and does not include header bytes
+func (store *Store) UpdateLiveUsedSpaceTotals(ctx context.Context, satelliteID storj.NodeID, pieceSize int64) {
+	store.liveUsedSpace.mu.Lock()
+	defer store.liveUsedSpace.mu.Unlock()
```

Comment:
Just curious what's the benefit of using defer than putting the `Unlock` after line 298

Reply:
in this case, i don't think it matters either way. I guess I just kept it like this for consistency since its how i use it in other places.

Reply:
How about in general? Is it because we want to keep the `Lock` and `Unlcok` close so we know for sure the `Unlock` is called?

Reply:
One reason to use `defer` in cases like this is that the defered function is still run when something down the call stack calls `panic()`. If we called `Unlock` normally at the end of the function, and there was a panic, the lock might never be unlocked. That hopefully isn't an issue here, because our code doesn't do much recovering from panics, but it could be elsewhere.

Reply:
I do think using defer to unlock makes that pretty clear. Also, for example, here https://github.com/storj/storj/pull/2707/files#diff-67139f1d1f58f7706b076d08baba9ec4R464 we return a value and want to unlock after so the defer helps with that. However, its best to minimize the time of locking, so you wouldn't want to use the defer to unlock if you don't need the lock for the rest of the function

Reply:
Thank you for explaining it! These are great info to know